appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java

```diff
@@ -1136,11 +1139,11 @@ public void onRemoved()
 
   public void delete() {
     this.editor.getProjectEditor().clearLocation(getName());
-    getForm().select(null);
     // Pass true to indicate that the component is being permanently deleted.
     getContainer().removeComponent(this, true);
     // tell the component its been removed, so it can remove children's blocks
     onRemoved();
+    getForm().select(null);
```

Comment:
Are we sure that moving this doesn't have any adverse side effects? I think it's okay but I'm not sure why it had to be moved.